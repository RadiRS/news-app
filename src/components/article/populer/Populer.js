import React from 'react';
import styled from 'styled-components';
import { Fonts, Images } from '../../../themes';
import { formatTime } from '../../../helpers';

const Populer = props => {
  const { multimedia, title, byline, published_date } = props.article;

  return (
    <Container>
      <Cover>
        {multimedia.length ? (
          <Image source={{ uri: multimedia[3].url }} />
        ) : (
          <Image source={{ uri: Images.noImage }} />
        )}
      </Cover>
      <Content>
        <Title numberOfLines={2}>{title}</Title>
        <Author>{byline}</Author>
        <Date>{formatTime(published_date)}</Date>
      </Content>
    </Container>
  );
};

export default Populer;

const Container = styled.View`
  background: white;
  width: 390px;
  height: 350px;
  border-radius: 14px;
  margin: 10px 10px 20px 10px;
  elevation: 10;
`;

const Cover = styled.View`
  width: 100%;
  height: 200px;
`;

const Image = styled.Image`
  border-top-left-radius: 14px;
  border-top-right-radius: 14px;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
`;

const Content = styled.View`
  padding: 20px;
`;

const Title = styled.Text`
  font-size: 24px;
  font-weight: bold;
`;

const Author = styled.Text`
  color: #b8bece;
  font-size: 15px;
  font-weight: ${Fonts.weight.large};
  margin-top: 2px;
`;

const Date = styled.Text`
  color: #b8bece;
  font-size: 15px;
  font-weight: ${Fonts.weight.large};
  margin-top: 2px;
`;
