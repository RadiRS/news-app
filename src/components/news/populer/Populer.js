import React from 'react';
import styled from 'styled-components';
import { Fonts, Images } from '../../../themes';

const Populer = props => {
  const { multimedia, title, byline, date } = props.article;

  // const renderImage = () => (
  //   multimedia.length ? (
  //     <Image source={{ uri: multimedia[3].url }} />
  //   ) : null
  // );
  return (
    <Container>
      {/* <Cover>{renderImage}</Cover> */}
      <Cover>
        {multimedia.length ? (
          <Image source={{ uri: multimedia[3].url }} />
        ) : (
          <Image source={{ uri: Images.noImage }} />
        )}
      </Cover>
      <Content>
        <Title numberOfLines={2}>{title}</Title>
        <Author>{byline}</Author>
        <Date>{date}</Date>
      </Content>
    </Container>
  );
};

export default Populer;

const Container = styled.View`
  background: white;
  width: 390px;
  height: 330px;
  border-radius: 14px;
  margin: 10px 10px 20px 10px;
  elevation: 10;
`;

const Cover = styled.View`
  width: 100%;
  height: 200px;
`;

const Image = styled.Image`
  border-top-left-radius: 14px;
  border-top-right-radius: 14px;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
`;

const Content = styled.View`
  padding: 20px;
`;

const Title = styled.Text`
  font-size: 24px;
  font-weight: bold;
`;

const Author = styled.Text`
  color: #b8bece;
  font-size: 15px;
  font-weight: ${Fonts.weight.large};
  margin-top: 2px;
`;

const Date = styled.Text`
  color: #b8bece;
  font-size: 15px;
  font-weight: ${Fonts.weight.large};
  margin-top: 2px;
`;
