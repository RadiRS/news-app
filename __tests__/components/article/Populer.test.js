import 'react-native';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import Populer from '../../../src/components/article/populer';

test('renders Populer component correctly', () => {
  const tree = renderer.create(<Populer article={dummyDataArticle} />).toJSON();
  expect(tree).toMatchSnapshot();
});

const dummyDataArticle = {
  title: 'Testing title',
  multimedia: [],
  byline: 'Radi Rasadi',
  published_date: '2019-04-08T18:06:10-04:00'
};
