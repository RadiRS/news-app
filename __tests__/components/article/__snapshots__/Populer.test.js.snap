// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Populer component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "borderRadius": 14,
        "elevation": 10,
        "height": 350,
        "marginBottom": 20,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "width": 390,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "height": 200,
          "width": "100%",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "http://lorempixel.com/640/480",
        }
      }
      style={
        Array [
          Object {
            "borderTopLeftRadius": 14,
            "borderTopRightRadius": 14,
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        },
      ]
    }
  >
    <Text
      numberOfLines={2}
      style={
        Array [
          Object {
            "fontFamily": "Domine Bold",
            "fontSize": 24,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Testing title
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#b8bece",
            "fontFamily": "Domine Bold",
            "fontSize": 15,
            "fontWeight": "600",
            "marginTop": 2,
          },
        ]
      }
    >
      Radi Rasadi
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#b8bece",
            "fontSize": 15,
            "fontWeight": "600",
            "marginTop": 2,
          },
        ]
      }
    />
  </View>
</View>
`;
